### DISCUSSION TITLE
Is there a way to filter operators by county?

### DISCUSSION THREAD

**Discussion by dmitrykinakh** [transitland/transitland#346] [Q&A] [answered]
Is there a way to have operators filtered by County name or State/Province?

For example, in v1 I was using this one call to have data filtered for a specific USA state.
```
https://api.transit.land/api/v1/operators?&apikey=API_KEY&limit=100&sort_key=id&sort_order=asc&state=US-NC&total=true
```

However, I don't see this field anymore in v2 - https://www.transit.land/documentation/rest-api/agencies#operator-request-parameters
Found only a case when I can specify county name in search parm but not sure if that is a correct one.
```
https://transit.land/api/v2/rest/operators?search=United%20States%20of%20America&sort_key=id&sort_order=asc&apikey=API_KEY
```
I'm looking for a correct way (to do not overload transit.land API with extra calls) how to get data loaded for all USA territory.

Also, I don't see anymore a field that was showing the total amount of records for a given query. Was that removed from API?
This is how summary information was represented in v1 of API if you add `total=true` to the API call.
```
{
	"operators": [],
	"meta": {
		"sort_key": "id",
		"sort_order": "asc",
		"per_page": 50,
		"offset": 0,
		"next": "https://api.transit.land/api/v1/operators?apikey=API_KEY&offset=50&per_page=50&sort_key=id&sort_order=asc",
		"total": 2798
	}
}
```

**Comment by dmitrykinakh** [comment]
@drewda - is this forum live at all?

**Reply by drewda** [reply]
Hi @dmitrykinakh thanks for your posts. Yes, we tend to this forum. Response time varies depending upon Interline staff's other tasks. Priority support is available by email on professional plans [#334](https://github.com/transitland/transitland/discussions/334#discussioncomment-1035158).

**Reply by dmitrykinakh** [reply]
So, more or less, you are saying that users with free usage of transit land might need to wait an undefined amount of time. That sad.
Our usage of transit land is not so high to buy a paid account. But I'm really disappointed @drewda about your answer. Since that means that you are not able to say when my question will be reviewed which hurts my project.

**Reply by rapsoj** [reply]
Any updates to this? @dmitrykinakh was a solution found?

**Reply by dmitrykinakh** [reply]
@rapsoj - well, we still stay on v1 since v2 is not suitable for us without the information I've requested above.

**Reply by dmitrykinakh** [reply]
@drewda - so after a while still no feedback to my question? In this case, what is the purpose of this forum?

**Reply by drewda** [reply]
@dmitrykinakh I am sorry to frustrate you, but we have not addressed this issue. You are right that the v2 REST API operator endpoint provides a consolidated `search` parameter. In the future, we will likely expand the query parameters and will provide updates at that point.

**Comment by drewda** [comment]
> Also, I don't see anymore a field that was showing the total amount of records for a given query. Was that removed from API?

Yes, the v2 REST API does not provide the total count of entities returned by a query. (Running these count queries in the v1 Datastore API can be very expensive and can impact other users, so we have removed this functionality.)

For some entities, there are other ways to get counts. For example, for entities that are imported from a feed, like stops and routes, there are statistics attached to the relevant FeedVersion: https://www.transit.land/documentation/rest-api/feed_versions

**Comment by drewda** [comment, marked as answer]
The Transitland v2 REST API now supports searching operators and agencies by country and state/province/division.

Here is the v2 equivalent of @dmitrykinakh's original query for operators in North Carolina:

```
https://transit.land/api/v2/rest/operators?adm1_iso=US-NC&apikey=API_KEY
```

See https://www.transit.land/documentation/rest-api/agencies for a longer list of query parameters.

Also note that Transitland is sourcing geographies, names, and ISO 3166 codes from the Natural Earth project: https://www.naturalearthdata.com/downloads/10m-cultural-vectors/

**Reply by sunjiayun2023** [reply]
It looks like this query only returns 20 operators. Is this by design or I made some mistakes in the request? Could you help with a way to return all operators (in North Carolina for example).

**Reply by drewda** [reply]
By default the operators endpoint includes 20 records per page. Either use the `after` parameter to advance to the next page: https://www.transit.land/documentation/rest-api/#pagination Or use the `limit` parameter to get more records on a single page.

In the case of North Carolina, there are only 21 operators total: https://www.transit.land/places/United%20States%20of%20America/North%20Carolina
